bundles/org.dataflowanalysis.analysis.pcm.informationflow/src/org/dataflowanalysis/analysis/pcm/informationflow/core/utils/IFConfidentialityVariableCharacterisationUtils.java

```diff
@@ -0,0 +1,508 @@
+			resolvedCharacterisations.add(copyConfidentialityVariableCharacterisation(characterisation));
+			return resolvedCharacterisations;
+		}
+
+		for (var characteristicType : allCharacteristicTypes) {
+			var resolvedConfChar = copyConfidentialityVariableCharacterisation(characterisation);
+
+			var lhs = getLhsEnumCharacteristicReference(resolvedConfChar);
+			lhs.setCharacteristicType(characteristicType);
+
+			List<NamedEnumCharacteristicReference> varsInRhs = extractAllVariables(resolvedConfChar.getRhs());
+			varsInRhs.stream().forEach(variable -> variable.setCharacteristicType(characteristicType));
+
+			resolvedCharacterisations.add(resolvedConfChar);
+		}
+		return resolvedCharacterisations;
+	}
+
+	private static ConfidentialityVariableCharacterisation copyConfidentialityVariableCharacterisation(
+			ConfidentialityVariableCharacterisation confChar) {
+
+		var lhs = getLhsEnumCharacteristicReference(confChar);
+		var characterisedVariable = confChar.getVariableUsage_VariableCharacterisation()
+				.getNamedReference__VariableUsage();
+
+		return createCharacterisation(characterisedVariable, lhs.getCharacteristicType(), lhs.getLiteral(),
+				copyTerm(confChar.getRhs()));
+	}
+
+	/**
+	 * Creates a {@link ConfidentialityVariableCharacterisation} for each level of
+	 * the given lattice for the given latticeCharacteristicType. The behavior of
+	 * the resulting {@link ConfidentialityVariableCharacterisation} is to set the
+	 * maximum Label of the given characterizations and the constraint. Assumes that
+	 * the given characterizations only set one level. Assumes that the given
+	 * characterizations do not have any wildcards. Assumes that only one
+	 * specification is present for each level - not more. Assumes at least one
+	 * characterization to be present.
+	 * 
+	 * @param confChars                 the given characterizations
+	 * @param constraint                the given constraint
+	 * @param latticeCharacteristicType the given CharacteristicType
+	 * @param lattice                   the given lattice
+	 * @return the resulting {@link ConfidentialityVariableCharacterisation} for the
+	 *         lattice
+	 */
+	public static List<ConfidentialityVariableCharacterisation> createModifiedCharacterisationsForAdditionalHigherEqualConstraint(
+			List<ConfidentialityVariableCharacterisation> confChars, AbstractNamedReference constraint,
+			CharacteristicType latticeCharacteristicType, Enumeration lattice) {
+
+		if (confChars.size() <= 0) {
+			String errorMsg = "At least one characterization has to be given.";
+			logger.error(errorMsg);
+			throw new IllegalArgumentException(errorMsg);
+		}
+		AbstractNamedReference characterizedVariable = confChars.get(0).getVariableUsage_VariableCharacterisation()
+				.getNamedReference__VariableUsage();
+
+		Map<Literal, ConfidentialityVariableCharacterisation> literalToDefinedConfCar = new HashMap<>();
+		for (var confChar : confChars) {
```

Comment:
Nitpick: Please avoid using abbreviations for variable names

Reply:
Changed